### DISCUSSION TITLE
Setting up development environment

### DISCUSSION THREAD

**Discussion by ghost** [apache/camel-karavan#920] [General]
Hello Guys, 

Can someone help me with setting up Karvan development environment ? I want to try making some changes also would like to contribute in future.

Thanks.

**Comment by ghost** [comment]
@mgubaidullin  Can you please suggest something here ?

**Comment by mgubaidullin** [comment]
1. Laptop with Linux or MacOS
2. Java 17, Maven 3.9, Jbang 0.110+
3. npm: 10+, node: 20
4. Docker 24+, minikube  v1.31+
5. VSCode 1.83 for extension development

The easiest way to build is to follow GitHub Actions for Karavan, ex: https://github.com/apache/camel-karavan/blob/main/.github/workflows/app.yml

**Reply by ghost** [reply]
Thanks @mgubaidullin 
That helped.
